### PR TITLE
br: Enable lint `gosec` in br

### PR DESCRIPTION
**Pull request by joccau**
Signed-off-by: joccau <zak.zhao@pingcap.com>

### What problem does this PR solve?
Enable `gosec` lints in br

Issue Number: close https://github.com/pingcap/tidb/issues/30699

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/mock/mock_cluster.go [hunk 1]
```diff
@@ -207,7 +207,7 @@ func waitUntilServerOnline(addr string, statusPort uint) string {
 	// connect http status
 	statusURL := fmt.Sprintf("http://127.0.0.1:%d/status", statusPort)
 	for retry = 0; retry < retryTime; retry++ {
-		resp, err := http.Get(statusURL) // nolint:noctx
+		resp, err := http.Get(statusURL) // nolint:gosec
```
Comment (kennytm) [inline br/pkg/mock/mock_cluster.go:210]: the `noctx` should still be kept?
```suggestion
		resp, err := http.Get(statusURL) //nolint:noctx,gosec
```

**Review by kennytm — APPROVED** [review]

**Review by fengou1 — APPROVED** [review]

**Comment by kennytm** [comment]
/run-mysql-test

```
[2021-12-27T04:25:30.279Z] time="2021-12-27T12:25:30+08:00" level=error msg="run test [auto_increment] err: sql:SHOW CREATE TABLE `t1`;: failed to run query \n\"SHOW CREATE TABLE `t1`;\" \n around line 443, \nwe need(325):\nSHOW CREATE TABLE `t1`;\nTable\tCreate Table\nt1\tCREATE TABLE `t1` (\n  `t1_name` varchar(255) DEFAULT NULL,\n  `t1_id` int(10) unsigned NOT NULL AUTO_INCREMENT,\n  PRIMARY KEY (`t1_id`) /*T![clustered_index] CLUSTERED */,\n  KEY `t1_name` (`t1_name`)\n) ENGINE=InnoDB DEFAULT CHARSET=latin1 COLLATE=latin1_bin AUTO_INCREMENT=2001000\nbut got(325):\nSHOW CREATE TABLE `t1`;\nTable\tCreate Table\nt1\tCREATE TABLE `t1` (\n  `t1_name` varchar(255) DEFAULT NULL,\n  `t1_id` int(10) unsigned NOT NULL AUTO_INCREMENT,\n  PRIMARY KEY (`t1_id`) /*T![clustered_index] CLUSTERED */,\n  KEY `t1_name` (`t1_name`)\n) ENGINE=InnoDB DEFAULT CHARSET=latin1 COLLATE=latin1_bin AUTO_INCREMENT=965339\n\n"
```

**Pull request merged**